Zend/Optimizer/zend_optimizer.c

```diff
@@ -847,11 +848,9 @@ const zend_class_constant *zend_fetch_class_const_info(
 			} else {
 				zend_class_entry *tmp = zend_hash_find_ptr(EG(class_table), Z_STR_P(op1 + 1));
 				if (tmp != NULL) {
-					if (tmp->type == ZEND_INTERNAL_CLASS) {
-						ce = tmp;
-					} else if (tmp->type == ZEND_USER_CLASS
-						&& tmp->info.user.filename
-						&& tmp->info.user.filename == op_array->filename) {
+					if (tmp->type == ZEND_INTERNAL_CLASS
+					 || (tmp->ce_flags & ZEND_ACC_PRELOADED)
+					 || (tmp->info.user.filename && tmp->info.user.filename == op_array->filename)) {
```

Comment:
Should this condition maybe become an inline function? Seems like it is repeated 4 times?

Reply:
Done